pkg/xds/generator/transparent_proxy_generator.go

```diff
@@ -37,16 +39,48 @@ func (_ TransparentProxyGenerator) Generate(ctx xds_context.Context, proxy *mode
 	if err != nil {
 		return nil, errors.Wrapf(err, "could not generate cluster: pass_through")
 	}
+
+	redirectPortInbound := proxy.Dataplane.Spec.Networking.GetTransparentProxying().GetRedirectPortInbound()
+	inboundPassThroughClusterName := "inbound:pass_through"
+	inboundPassThroughCluster, err := envoy_clusters.NewClusterBuilder().
+		Configure(envoy_clusters.PassThroughCluster(inboundPassThroughClusterName)).
+		Configure(envoy_clusters.UpstreamBindConfig("127.0.0.6", 0)).
+		Build()
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not generate cluster: %s", inboundPassThroughClusterName)
+	}
+
+	inboundListenerName := names.GetInboundListenerName("0.0.0.0", redirectPortInbound)
+	inboundListener, err := envoy_listeners.NewListenerBuilder().
+		Configure(envoy_listeners.InboundListener(inboundListenerName, "0.0.0.0", redirectPortInbound)).
+		Configure(envoy_listeners.FilterChain(envoy_listeners.NewFilterChainBuilder().
+			Configure(envoy_listeners.TcpProxy(inboundPassThroughClusterName, envoy_common.ClusterSubset{ClusterName: inboundPassThroughClusterName})))).
+		Configure(envoy_listeners.OriginalDstForwarder()).
+		Build()
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not generate listener: %s", inboundListenerName)
+	}
+
 	return []*model.Resource{
-		&model.Resource{
+		{
 			Name:     "catch_all",
 			Version:  proxy.Dataplane.Meta.GetVersion(),
 			Resource: listener,
 		},
-		&model.Resource{
+		{
 			Name:     "pass_through",
 			Version:  proxy.Dataplane.Meta.GetVersion(),
 			Resource: cluster,
 		},
```

Comment:
Can we get here consistent naming for passthrough clusters? I was thing of something about:
`inbound:passthrough` (for listener and cluster)
`outbound:passthrough` (for listener and cluster)
WDYT?

Reply:
Yes, I agree. For listeners it also could be standard format: `inbound:0.0.0.0:15006` and `outbound:0.0.0.0:15001` to be more descriptive

Reply:
Actually will the same names for listener and cluster work? We store it in a ResourceSet, right?

Reply:
It should work because they are unique per type. I think I like this one:
```
listener - inbound:passthrough
cluster - inbound:passthrough

listener - outbound:passthrough
cluster - outbound:passthrough
```
because when browsing stats in Grafana `inbound:passthrough ` tells me more than `inbound:0.0.0.0:15006`